codemod/base.py

```diff
@@ -0,0 +1,1000 @@
+
+    parser.add_argument('--accept-all', action='store_true',
+                        help='Automatically accept all '
+                             'changes (use with caution).')
+
+    parser.add_argument('--default-no', action='store_true',
+                        help='If set, this will make the default '
+                             'option to not accept the change.')
+
+    parser.add_argument('--editor', action='store', type=str,
+                        help='Specify an editor, e.g. "vim" or emacs". '
+                        'If omitted, defaults to $EDITOR environment '
+                        'variable.')
+    parser.add_argument('--count', action='store_true',
+                        help='Don\'t run normally.  Instead, just print '
+                             'out number of times places in the codebase '
+                             'where the \'query\' matches.')
+    parser.add_argument('--test', action='store_true',
+                        help='Don\'t run normally.  Instead, just run '
+                             'the unit tests embedded in the codemod library.')
+
+    parser.add_argument('match', nargs='?', action='store', type=str,
+                        help='Regular expression to match.')
+    parser.add_argument('subst', nargs='?', action='store', type=str,
+                        help='Substitution to replace with.')
+
+    arguments = parser.parse_args()
+
+    if arguments.test:
+        import doctest
+        doctest.testmod()
+        sys.exit(0)
+
+    if (
+            arguments.extensions is None
+    ) and (arguments.include_extensionless is False):
+        parser.print_usage()
+        sys.exit(0)
+
+    yes_to_all = arguments.accept_all
+
+    query_options = {}
+
+    query_options['suggestor'] = (
+        multiline_regex_suggestor if arguments.m else regex_suggestor
+    )(arguments.match, arguments.subst, arguments.i)
+
+    query_options['start'] = arguments.start
+    query_options['end'] = arguments.end
+    query_options['root_directory'] = arguments.d
+    query_options['inc_extensionless'] = arguments.include_extensionless
+
+    if arguments.extensions is not None or arguments.exclude_paths is not None:
+        query_options['path_filter'] = (
+            path_filter(extensions=arguments.extensions.split(',')
+                        if arguments.extensions is not None else None,
+                        exclude_paths=arguments.exclude_paths.split(',')
+                        if arguments.exclude_paths is not None else None))
+
+    options = {}
+    options['query'] = Query(**query_options)
+    if arguments.editor is not None:
+        options['editor'] = arguments.editor
+    options['just_count'] = arguments.count
+    options['default_no'] = arguments.default_no
+
+    return options
+
+
+def main():
+    options = _parse_command_line()
+    run_interactive(**options)
+
+if __name__ == '__main__':
+    main()
```

Comment:
It's so great that you turned this script into a package. I hope in future commits we can start splitting up this monolithic file.